test/toolbox_image.sh

```diff
@@ -254,36 +263,124 @@ test_system_update_entrypoint()
 
     # Remove the identifiers in the header because they will always change.
     sed -i "s/label-id:.*//" "${rootfs_dir}${PARTITION_TABLE_FILE}"
     sed -i "s/label-id:.*//" "${rootfs_dir}${verification_partition_table_file}"
     diff "${rootfs_dir}${PARTITION_TABLE_FILE}" "${rootfs_dir}${verification_partition_table_file}" || return 1
 }
 
+test_execute_disk_prepare_grow_boot_overlapping_rootfs_nok()
+{
+    new_boot_size="$((ROOTFS_START - BOOT_START + 1))"
+
+    sed -i "s/p2.*type/p1 : start=     ${BOOT_START}, size=     ${new_boot_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 0
+}
+
+test_execute_disk_prepare_grow_rootfs_overlapping_userdata_nok()
+{
+    new_rootfs_size="$((USERDATA_START - ROOTFS_START + 1))"
+
+    sed -i "s/p2.*type/p2 : start=     ${ROOTFS_START}, size=     ${new_rootfs_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 0
+}
+
+test_execute_disk_prepare_grow_boot_invalid_start_nok()
+{
+    new_boot_start="2047"
+    new_boot_size="$((ROOTFS_START - new_boot_start))"
+
+    # In every line in the partition table look for a string "p1<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p1.*type/p1 : start=     ${new_boot_start}, size=     ${new_boot_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 0
+}
+
+test_execute_disk_prepare_grow_beyond_disk_end_nok()
+{
+    new_userdata_size="$((STORAGE_DEVICE_END - ROOTFS_START))"
+
+    # In every line in the partition table look for a string "p3<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p3.*type/p3 : start=     ${USERDATA_START}, size=     ${new_userdata_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 0
+}
+
+test_execute_resize_partition_grow_rootfs_ok()
+{
+    userdata_size="$((STORAGE_DEVICE_SIZE - USERDATA_START))"
+    new_userdata_size="$((userdata_size / 2))"
+    new_userdata_start="$((USERDATA_START + new_userdata_size))"
+    new_rootfs_size="$((new_userdata_start - ROOTFS_START))"
+    # In every line in the partition table look for a string "p3<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s/p2.*type/p2 : start=       ${ROOTFS_START}, size=     ${new_rootfs_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+    sed -i "s/p3.*type/p3 : start=     ${new_userdata_start}, size=     ${new_userdata_size}, type/" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    helper_execute_prepare_disk || return 1
+    helper_test_disk_integrity || return 1
+}
+
+test_execute_disk_prepare_grow_boot_ok()
+{
+    rootfs_size="$((USERDATA_START - ROOTFS_START))"
+    new_rootfs_size="$((rootfs_size / 2))"
```

Comment:
while this is a good test, don't we want to maybe grow the rootfs as well? IN any case the function name doens't fully map to this. You only want to grow the boot. Granted to be able to do that somethings got to give.

Reply:
The test function says "test execution of disk_prepare by growing boot, expected result ok. The input is a partition table that should be correct, therefore the function modifies the boot "and" the rootfs partitions.